utils/actions/detectConsoleLogs.ts

```diff
@@ -109,63 +84,49 @@ export default async function detectConsoleLogs({
   const commentPromises = diffFiles.map(async (file) => {
     const { additions } = getLineDiffs(file.patch ?? "");
 
-    // detect if the additions contain console logs or not
-    try {
-      return await openai
-        .createChatCompletion({
-          model: "gpt-4-1106-preview",
-          messages: [
-            {
-              role: "system",
-              content: `${consoleLogDetectionPrompt} \n ${additions}`,
-            },
-          ],
-        })
-        .then((result) => {
-          const openAIResult =
-            result.data.choices[0].message.content.split(",");
-
-          const addtionsHaveConsoleLog = openAIResult[0];
-          const individualLine = openAIResult[1];
-
-          if (addtionsHaveConsoleLog === "true") {
-            const commentFileDiff = () => {
-              const consoleLogPosition = getConsoleLogPosition({
-                filePatch: file.patch ?? "",
-                individualLine,
-              });
-
-              return octokit
-                .request(
-                  "POST /repos/{owner}/{repo}/pulls/{pull_number}/reviews",
-                  {
-                    owner,
-                    repo,
-                    pull_number: issue_number,
-                    commit_id:
-                      typeof latestCommitHash === "string"
-                        ? latestCommitHash
-                        : undefined,
-                    event: "COMMENT",
-                    path: file.filename,
-                    comments: [
-                      {
-                        path: file.filename,
-                        position: consoleLogPosition || 1, // comment at the beggining of the file by default
-                        body: commentBody,
-                      },
-                    ],
-                  }
-                )
-                .catch((err) => {
-                  console.log(err);
-                });
-            };
-
-            commentFileDiff();
-          }
-        });
-    } catch {}
+    const splitAdditions = additions.split("\n");
+
+    // RegEx to detect console log equivalents in different languages
+    const consoleLogRegex = /(console\.log|print|printf|fmt\.Print|log\.Print|NSLog|puts|println|println!)\([^)]*\)(?![^]*?\/\/|[^]*?\/\*|#)/;
+    
+    // RegEx to ignore lines that contian //, /*, etc.
+    const commentRegex = /^\s*\/{2,}|^\s*\/\*|\*\/|#/;
+
+    for (let i=0; i < splitAdditions.length; i++) {
+      let currentLine = splitAdditions[i];
+      if (!currentLine.match(commentRegex) && currentLine.match(consoleLogRegex)) {
+        const commentFileDiff = async () => {
+
+          const consoleLogPosition = i + 1; // The +1 is because IDEs and GitHub file diff view index LOC at 1, not 0
```

Comment:
I know this is fast, but should we do it concurrently?

Reply:
Also, this generates a comment in the _last_ consle log, isn't it better to show it at the top of the PR?

Reply:
Steps to replicate? 

I'm testing with this fizz buzz and it's commenting lines 6,8, and 10 respectively

<img width="511" alt="Screen Shot 2023-12-14 at 11 16 54 AM" src="https://github.com/watermelontools/watermelon/assets/8325094/154d6bee-fa36-4e4d-ac30-fee77453f1fe">

Reply:
For really long file diffs it will be a long running process, awaiting each time the posting of the comment.
As per the [API](https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#create-a-review-for-a-pull-request) we can send an object and never hit the rate limit.

Reply:
In the very rare edge case that we get a very long file diff where every single line should be commented, I still don't think we'd have problems. 

My argument for this is that when doing this with GPT, we were doing the very same thing with the additional computational complexity of async/awaiting the http call to Open AI which is an expensive one. 

If we ever get to the rate limit we can always refactor this.